bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/ScriptEngineManagerImpl.java

```diff
@@ -163,6 +166,17 @@ public void loadScript(String engineIdentifier, InputStreamReader scriptData) {
         }
     }
 
+    private void tryExtractAndInjectFilename(String engineIdentifier, ScriptEngine engine) {
+        try {
+            File maybeScriptFile = new File(engineIdentifier);
+            if(maybeScriptFile.isFile()) {
+                engine.getContext().setAttribute(ScriptEngine.FILENAME, maybeScriptFile.getName(), ScriptContext.ENGINE_SCOPE);
+            }
+        } catch (Exception e) {
+            logger.warn("Failure whilst inserting script filename: {}" + e.getMessage());
```

Comment:
> logger.warn("Failure whilst inserting script filename: {}" + e.getMessage());

    logger.warn("Failure while adding script '{}' file name to the ScriptEngine: {}", engineIdentifier, e.getMessage());

This is a little more descriptive and utilizes parameterization.